### PR TITLE
Check if blob is ready for processing

### PR DESCRIPTION
**Pull request by rkondratowicz**
Using checker class from previous PR in the blob processing job.

Best to review with whitespace changes **OFF**.

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessorTest.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import java.util.Optional;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+class BlobProcessorTest {
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessorTest.java]: note that unit tests for `BlobProcessor` did not exist before, but adding any other relevant tests for this class is not in the scope of this PR.

Only doing checks around blob readiness.

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessorTest.java [hunk 2]
```diff
@@ -56,7 +62,15 @@ protected void beforeTest() {
...
-        blobProcessor = new BlobProcessor(storageClient, dispatcher, envelopeRepository, leaseClientProvider);
+        when(readinessChecker.isReady(any())).thenReturn(true); // disable processing delay.
+
+        blobProcessor = new BlobProcessor(
+            storageClient,
+            dispatcher,
+            readinessChecker,
+            envelopeRepository,
+            leaseClientProvider
+        );
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessorTest.java]: line got too long with the new parameter, sorry.

File: src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessor.java [hunk 3]
```diff
@@ -64,24 +68,32 @@ private void processBlob(String blobName, String containerName) {
...
-                blobClient.getProperties().getCreationTime().toInstant()
-            ));
-
-            logger.info(
-                "Finished processing {} from {} container. New envelope ID: {}",
-                blobName,
-                containerName,
-                envelopeId
-            );
+            if (this.readinessChecker.isReady(blobClient.getProperties().getCreationTime().toInstant())) {
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessor.java]: We could assign creation time to local variable as it is used twice in the same scope. Will also make if statement a little bit more sensible

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessor.java]: good point

**Pull request merged**